### PR TITLE
yder: init at 1.4.5

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
###### Motivation for this change
Is in lots of other repos: https://repology.org/project/yder/versions

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested compilation of all library files
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/orcania/default.nix [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  version = "2.0.0";
+
+  src = fetchFromGitHub {
+    owner = "babelouest";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "11ihbfm7qbqf55wdi7azqx75ggd3l0n8ybyq2ikidffvmg13l4g9";
+  };
+
+  nativeBuildInputs = [
```
Comment (markuskowa) [inline pkgs/development/libraries/orcania/default.nix]: That statement fits in one line

File: pkgs/development/libraries/orcania/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    cmake
+  ];
+
+  meta = {
```
Comment (markuskowa) [inline pkgs/development/libraries/orcania/default.nix]: a `with stdenv.lib;` here make the rest below more readable.

File: pkgs/development/libraries/yder/default.nix [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    sha256 = "0j46v93vn130gjcr704rkdiibbk3ampzsqb6xdcrn4x115gwyf5i";
+  };
+
+  patches = [
+    # We set CMAKE_INSTALL_LIBDIR to the absolute path in $out, so
+    # prefix and exec_prefix cannot be $out, too
+    ./fix-pkgconfig.patch
+  ];
+
+  nativeBuildInputs = [
```
Comment (markuskowa) [inline pkgs/development/libraries/yder/default.nix]: Make it maybe one line? Same for `checkInputs` and `cmakeFlags`

File: pkgs/development/libraries/yder/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    "-DBUILD_YDER_TESTING=on"
+  ];
+
+  doCheck = true;
+
+  preCheck = ''
+    export LD_LIBRARY_PATH="$(pwd):$LD_LIBRARY_PATH"
+  '';
+
+  meta = {
```
Comment (markuskowa) [inline pkgs/development/libraries/yder/default.nix]: with stdenv.lib;

File: pkgs/development/libraries/yder/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    description = "Logging library for C applications";
+    homepage = "https://github.com/babelouest/yder";
+    license = stdenv.lib.licenses.lgpl21;
+    maintainers = with stdenv.lib.maintainers; [ johnazoidberg ];
```
Comment (markuskowa) [inline pkgs/development/libraries/yder/default.nix]: Since the packages relies on `systemd` it should set `platforms` to `linux` only.

**Comment by JohnAZoidberg** [comment]
What about this?
We can build it without systemd on non-linux. :)

**Comment by markuskowa** [comment]
@GrahamcOfBorg build yder

**Comment by markuskowa** [comment]
> What about this?
> We can build it without systemd on non-linux. :)

Good idea. Unfortunately it fails to build on darwin.

**Comment by JohnAZoidberg** [comment]
Let's hope this fixes it :D
@GrahamcOfBorg build yder

**Comment by JohnAZoidberg** [comment]
Oh, can you please have ofBorg run it on Darwin? @markuskowa 
I'm not a privileged user so it doesn't build there.

File: pkgs/development/libraries/yder/fix-darwin.patch [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
++    target_compile_definitions(yder_static PUBLIC -DO_STATIC_LIBRARY)
++    set_target_properties(yder_static PROPERTIES
++            OUTPUT_NAME yder)
++endif ()
++
+ # build yder-cfg.h file
+ configure_file(${INC_DIR}/yder-cfg.h.in ${PROJECT_BINARY_DIR}/yder-cfg.h)
+ set (CMAKE_EXTRA_INCLUDE_FILES ${PROJECT_BINARY_DIR})
+diff --git i/Makefile w/Makefile
+deleted file mode 100644
```
Comment (markuskowa) [inline pkgs/development/libraries/yder/fix-darwin.patch]: If understand it correctly from here on your patch just deletes a bunch of make files. Why not using `rm Makefile` etc in `postPatch`? That would be shorter and more clear.

**Comment by markuskowa** [comment]
Still fails on darwin.

File: pkgs/development/libraries/yder/fix-darwin.patch (hunk 6 above)
Comment (JohnAZoidberg) [reply pkgs/development/libraries/yder/fix-darwin.patch]: Oh, those deletions weren't supposed to be included - that was from some experimentation that I did. The main change is of that in `CMakeLists.txt`.

**Comment by JohnAZoidberg** [comment]
Yes, still fails, but now during linking the test executable:
```
[100%] Linking C executable yder_test
/nix/store/37786hp65ar94pkml5yv40xa6gmjbkx0-cmake-3.13.4/bin/cmake -E cmake_link_script CMakeFiles/yder_test.dir/link.txt --verbose=y
/nix/store/2p5k5g7byv3vnswwlj5gramgrh31zdgm-clang-wrapper-7.1.0/bin/clang  -Wall -Werror -O3 -DNDEBUG -arch x86_64 -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/yder_test.dir/test/yder_test.c.o  -o yder_test libyder.1.4.6.dylib /nix/store/ggyq1118bh543dxzfab1bfd1d6zvs626-orcania-2.0.0/lib/liborcania.dylib /nix/store/lv0sjs2f6i3hidphrb2ijnzx1hjm0wwr-check-0.12.0/lib/libcheck.dylib /nix/store/7jmka06vi7mjsgg9i50lgw649azixb8m-subunit-1.1.0/lib/libsubunit.dylib -lm -lrt
ld: library not found for -lrt
```

**Comment by JohnAZoidberg** [comment]
Alright, I think I'm close. (Thanks for rebuilding :))
```
    Start 1: yder_test
1/1 Test #1: yder_test ........................Child aborted***Exception:   0.00 sec
dyld: Library not loaded: libyder.1.4.dylib
  Referenced from: /tmp/nix-build-yder-1.4.6.drv-0/source/build/yder_test
  Reason: image not found
0% tests passed, 1 tests failed out of 1
```
Don't know why it works on Linux but not all libraries are linked to the test binary.

**Comment by markuskowa** [comment]
Still the same problem.

**Comment by JohnAZoidberg** [comment]
Even more fixes:
- Test for journald logging is not systemd specific and fails on darwin
- If the working directory is different from the build directory, the  
  dynamic linker can't find libyder

**Comment by markuskowa** [comment]
You're sure you don't want to restrict it to Linux only? Without a mac it is pretty hard to test. In the end we might make it compile but that is no guarantee that it also runs reliably.

**Comment by JohnAZoidberg** [comment]
Yeah, you're right. But I'm so close ;) Got all the linker errors fixed and now only the test is remaining. I think I fixed that now.

I'll split my fixes and upstream those patches ASAP.
When it at least builds on darwin, other people can provide upstream with feedback whether it reliably works. Maybe even Nix on Darwin users.

**Comment by markuskowa** [comment]
Looks like it works now.

**Pull request merged**